examples/long_running_measurement/measurement.py

```diff
@@ -0,0 +1,103 @@
+"""Perform a long running measurement."""
+import logging
+import pathlib
+import random
+import threading
+import time
+from typing import Generator, List, Tuple
+
+import click
+import grpc
+
+import ni_measurementlink_service as nims
+
+service_directory = pathlib.Path(__file__).resolve().parent
+measurement_service = nims.MeasurementService(
+    service_config_path=service_directory / "LongRunningMeasurement.serviceconfig",
+    version="0.5.0.0",
+    ui_file_paths=[service_directory / "LongRunningMeasurement.measui"],
+)
+
+
+Outputs = Tuple[List[int], str, List[int], float]
+
+
+@measurement_service.register_measurement
+@measurement_service.configuration("time_in_seconds", nims.DataType.Int32, 10)
+@measurement_service.output("random_numbers", nims.DataType.Int32Array1D)
+@measurement_service.output("status", nims.DataType.String)
+@measurement_service.output("overlimit", nims.DataType.Int32Array1D)
+@measurement_service.output("elapsed_time", nims.DataType.Double)
+def measure(time_in_seconds: float) -> Generator[Outputs, None, Outputs]:
+    """Perform a long running measurement."""
+    logging.info("Executing measurement: time_in_seconds=%d", time_in_seconds)
+    cancellation_event = threading.Event()
+    measurement_service.context.add_cancel_callback(cancellation_event.set)
+
+    status = ""
+    random_numbers: List[int] = []
+    overlimit: List[int] = []
+    elapsed_time = 0.0
+
+    start_time = time.monotonic()
+    stop_time = start_time + time_in_seconds
+    loop_index = 0
+    while time.monotonic() < stop_time:
+        status = f"Loop index: {loop_index}"
+        loop_index += 1
+
+        elapsed_time = time.monotonic() - start_time
+
+        desired_len = int(elapsed_time * 1000.0 / 10.0)
+        while len(random_numbers) < desired_len:
+            random_number = random.randint(1, 1000)
+            random_numbers.append(random_number)
+            if random_number > 800:
```

Comment:
I think the limit should be an input to the measurement. At a minimum, I think there should be a comment on the UI that the limit is hardcoded to 800.

Reply:
I'll post a separate PR with the revised example.